File: generated-code/codecraft/csharp/Model/EntityType.cs
namespace TransGenTest.Model
{
    public enum EntityType
    {
        Wall = 0,
        House = 1,
        BuilderBase = 2,
        BuilderUnit = 3,
        MeleeBase = 4,
        MeleeUnit = 5,
        RangedBase = 6,
        RangedUnit = 7,
        Resource = 8,
        Turret = 9,
    }

    public static class EntityTypeHelper {
        public static EntityType ReadFrom(System.IO.BinaryReader reader) {
            switch (reader.ReadInt32())
            {
                case 0:
                    return EntityType.Wall;
                case 1:
                    return EntityType.House;
                case 2:
                    return EntityType.BuilderBase;
                case 3:
                    return EntityType.BuilderUnit;
                case 4:
                    return EntityType.MeleeBase;
                case 5:
                    return EntityType.MeleeUnit;
                case 6:
                    return EntityType.RangedBase;
                case 7:
                    return EntityType.RangedUnit;
                case 8:
                    return EntityType.Resource;
                case 9:
                    return EntityType.Turret;
                default:
                    throw new System.Exception("Unexpected tag value");
            }
        }
    }
}

File: generated-code/codecraft/go/model/entity_type.go
package model

import "io"
import . "trans_gen_test/stream"

// Entity type
type EntityType int32

const (
    // Wall, can be used to prevent enemy from moving through
    EntityTypeWall EntityType = 0
    // House, used to increase population
    EntityTypeHouse EntityType = 1
    // Base for recruiting new builder units
    EntityTypeBuilderBase EntityType = 2
    // Builder unit can build buildings
    EntityTypeBuilderUnit EntityType = 3
    // Base for recruiting new melee units
    EntityTypeMeleeBase EntityType = 4
    // Melee unit
    EntityTypeMeleeUnit EntityType = 5
    // Base for recruiting new ranged units
    EntityTypeRangedBase EntityType = 6
    // Ranged unit
    EntityTypeRangedUnit EntityType = 7
    // Resource can be harvested
    EntityTypeResource EntityType = 8
    // Ranged attacking building
    EntityTypeTurret EntityType = 9
)

// Read EntityType from reader
func ReadEntityType(reader io.Reader) EntityType {
    switch ReadInt32(reader) {
    case 0:
        return EntityTypeWall
    case 1:
        return EntityTypeHouse
    case 2:
        return EntityTypeBuilderBase
    case 3:
        return EntityTypeBuilderUnit
    case 4:
        return EntityTypeMeleeBase
    case 5:
        return EntityTypeMeleeUnit
    case 6:
        return EntityTypeRangedBase
    case 7:
        return EntityTypeRangedUnit
    case 8:
        return EntityTypeResource
    case 9:
        return EntityTypeTurret
    }
    panic("Unexpected tag value")
}

// Get string representation of EntityType
func EntityTypeToString(entityType EntityType) string {
    switch entityType {
    case EntityTypeWall:
        return "Wall"
    case EntityTypeHouse:
        return "House"
    case EntityTypeBuilderBase:
        return "BuilderBase"
    case EntityTypeBuilderUnit:
        return "BuilderUnit"
    case EntityTypeMeleeBase:
        return "MeleeBase"
    case EntityTypeMeleeUnit:
        return "MeleeUnit"
    case EntityTypeRangedBase:
        return "RangedBase"
    case EntityTypeRangedUnit:
        return "RangedUnit"
    case EntityTypeResource:
        return "Resource"
    case EntityTypeTurret:
        return "Turret"
    }
    panic("Impossible happened")
}